### PR TITLE
Gradle: Convert to Gradle build scripts.

### PR DESCRIPTION
**Pull request by lijamie98**
<details>
  <summary>Convert to Gradle build script</summary>

### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

Migrate from Maven build to Gradle

### Why

Maven is slow, complicated and hard to maintain. 

### Known limitations

This is still a monolithic Gradle script. We will create multiple projects in the coming commits.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,74 +1,3 @@
-# These are some examples of commonly ignored file patterns.
```
Comment (marcelosalloum) [inline .gitignore:1]: Why are you removing the gitignore?

**Review by marcelosalloum — COMMENTED** [review]
Can you also update the Github Action `build_and_test.yml` to run gradle tests instead of maven?
https://github.com/stellar/java-stellar-anchor-sdk/blob/f949650d40233aff82c40dd02c48d2422ab3f2e0/.github/workflows/build_and_test.yml#L23-L29

The tests are failing.

**Review by JakeUrban — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (lijamie98) [reply .gitignore:1]: These are not necessary, They were from the legacy code. I cleaned them up for maintainability.

File: settings.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+rootProject.name = "java-stellar-anchor-sdk"
+
+enableFeaturePreview("VERSION_CATALOGS")
+
+dependencyResolutionManagement {
```
[the remark this answers is not in the document]
Comment (lijamie98) [reply settings.gradle.kts:5]: This is the version catalog feature of the Gradle's dependency management. 

https://docs.gradle.org/current/userguide/platforms.html

**Pull request merged**